src/twisted/conch/test/test_manhole.py

```diff
@@ -244,10 +242,21 @@ def finished(ign):
                     + defaultFunctionName.encode("utf-8"),
                     b"Exception: foo bar baz",
                     b">>> done",
-                ]
+                ],
             )
 
-        return done.addCallback(finished)
+        done.addCallback(finished)
+        return done
+
+    def test_ExceptionWithCustomExcepthook(
+        self,
+    ):
+        """
+        Raised exceptions are handled the same way even if L{sys.excepthook}
+        has been modified from its original value.
+        """
+        self.patch(sys, "excepthook", lambda *args: None)
```

Comment:
Maybe something like this.
This should make coverage happy and should also help with the docs.

There is an exception in the coveragerc config file for `raise NotImplementedError`

This is just a suggetion/idea and might not work,

```suggestion
        def never_called(*args):
           """
           Coverage will complain that this is not called.
           but that is the purpose of this code.
           It is placeholder for the global sys.excepthook
           and the test is here to make sure the global sys.excepthook
           is not called.
           """
           raise NotImplementedError("never_called shouldn't be called")

        self.patch(sys, "excepthook", never_called)
```